### ISSUE TITLE
aws_eks_node_group  desired_size is always 1

### ISSUE THREAD

**Issue by FernandoMiguel** [hashicorp/terraform-provider-aws#25486]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
```
Terraform v1.2.3
on darwin_arm64
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/aws v4.19.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/helm v2.6.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.11.0
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.3.1
+ provider registry.terraform.io/hashicorp/time v0.7.2
+ provider registry.terraform.io/hashicorp/vault v3.7.0
```

### Affected Resource(s)

* aws_eks_node_group

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/85e5cae702bafcb81e7e74ec1497cda772de4b86/modules/eks-managed-node-group/main.tf#L283)

### Debug Output
```
      ~ eks_managed_node_groups                                    = {
          ~ spot_node_group = {
              + FOO_aws_eks_node_group             = [
                  + {
                      + ami_type               = "AL2_x86_64"
                      + arn                    = "arn:aws:eks:us-east-1:XXX:nodegroup/28b590cd8624"
                      + capacity_type          = "SPOT"
                      + cluster_name           = "kingfish"
                      + disk_size              = 100
                      + force_update_version   = true
                      + id                     = "kingfish-ng"
                      + instance_types         = [
                          + "c6a.large",
                          + "c6i.large",
                        ]
                      + launch_template        = []
                      + node_group_name        = "ng"
                      + node_group_name_prefix = ""
                      + node_role_arn          = "arn:aws:iam::XXX:role/397d9cfc"
                      + release_version        = "1.22.9-20220610"
                      + remote_access          = []
                      + resources              = [
                          + {
                              + autoscaling_groups              = [
                                  + {
                                      + name = "eks28b590cd8624"
                                    },
                                ]
                              + remote_access_security_group_id = ""
                            },
                        ]
                      + scaling_config         = [
                          + {
                              + desired_size = 1
                              + max_size     = 6
                              + min_size     = 2
                            },
```

### Expected Behavior
setting `desired_size=3` would launch 3 instances

### Actual Behavior
the scaling options sets `desired_size` to 1
```
                      + scaling_config         = [
                          + {
                              + desired_size = 1
                              + max_size     = 6
                              + min_size     = 2
                            },
                        ]
```

**Comment by StephanX** [comment]
I couldn't quite find where in the plugin code the problem originates, but you can mitigate it (at least temporarily) by editing `.terraform/eks_node_group/main.tf` and commenting out line 112: `# ignore_changes        = [scaling_config[0].desired_size]` ; basically, the desired_size is being set on cluster creation, then ignored subsequently.

**Comment by mattburgess** [comment]
@FernandoMiguel are you happy for this to be closed (either outright, or as a duplicate of #21668)? As @StephanX mentions, the behaviour you're seeing is expected given that the module you're using explicitly ignores changes to the `desired_size` attribute (https://github.com/terraform-aws-modules/terraform-aws-eks/blob/cc3fe0590b6d7c8bf2a1690c5331fa92527efbe4/modules/eks-managed-node-group/main.tf#L340-L345).

**Comment by FernandoMiguel** [comment]
Yes.
